pkg/controller/apis/config/v1alpha1/defaults.go

```diff
@@ -62,6 +65,44 @@ func SetDefaults_KubeControllerManagerConfiguration(obj *kubectrlmgrconfigv1alph
 
 	// Use the default RecommendedDefaultGenericControllerManagerConfiguration options
 	RecommendedDefaultGenericControllerManagerConfiguration(&obj.Generic)
+	// Use the default RecommendedDefaultHPAControllerConfiguration options
+	attachdetachconfigv1alpha1.RecommendedDefaultAttachDetachControllerConfiguration(&obj.AttachDetachController)
```

Comment:
these controller-specific defaults should just be normal defaulting functions so we don't have to remember to call them in every place they are used. opened https://github.com/kubernetes/kubernetes/issues/76027 to track changing these to normal defaulting functions.